include/sqlpp11/result.h

```diff
@@ -93,25 +93,39 @@ namespace sqlpp
       using reference = const result_row_t&;
       using difference_type = std::ptrdiff_t;
 
-      iterator(std::reference_wrapper<db_result_t> result,
-               std::reference_wrapper<result_row_t> result_row) :
-          _result(std::move(result)), _result_row(std::move(result_row))
+      iterator()
+          : _result_ptr(nullptr), _result_row_ptr(nullptr)
+      {
+      }
+
+      iterator(db_result_t& result, result_row_t& result_row)
+          : _result_ptr(&result), _result_row_ptr(&result_row)
       {
       }
 
       reference operator*() const
       {
-        return _result_row;
+        return *_result_row_ptr;
       }
 
       pointer operator->() const
       {
-        return &_result_row.get();
+        return _result_row_ptr;
       }
 
       bool operator==(const iterator& rhs) const
       {
-        return _result_row.get() == rhs._result_row.get();
+        if ((_result_row_ptr != nullptr) != (rhs._result_row_ptr != nullptr))
+        {
+          return false;
+        }
+        if (! _result_row_ptr)
+        {
+          return true;
+        }
+        // FIXME: Throw an exception when someone tries to compare iterators from different sequences,
+        // because this is undefined behavior.
```

Comment:
"undefined behavior" means undefined behavior, not "throw an exception" or "crash" or "abort" or anything.

Returning a result like this is perfectly fine, afaict.

Reply:
OK, I removed the comment.

Indeed it is perfectly fine to do anything on undefined behavior including the current action. However I think that comparing iterators from different sequences is clearly a programming mistake and it is difficult to imagine a scenario where it would be done intentionally. That is why I added the comment.